### ISSUE TITLE
Feature request: integer to ScanCode casting?

### ISSUE THREAD

**Issue by kkimdev** [rsaarelm/scancode-rs#1]
How about having an integer to ScanCode casting function, that returns `Result<ScanCode, ()>`.

For me, I needed for converting a scancode returned from SDL C library.

Some relevant links:
- https://www.reddit.com/r/rust/comments/36pgn9/integer_to_enum_after_removal_of_fromprimitive/
- https://github.com/andersk/enum_primitive-rs

**Comment by rsaarelm** [comment]
Can't you use `Scancode::new` for this?

**Comment by kkimdev** [comment]
Wait, isn't that for a totally different type of conversion?  For example, I'm expecting `Scancode::new(4) == Scancode::A`.

**Comment by rsaarelm** [comment]
It converts scancode values from your local system, which should be what SDL events carry, into values of the crate's enum. So `Scancode::new(4)` would be `Some(Scancode::Num3)` if you're on Windows and `None` if you're on Linux.

The values in the actual `Scancode` enum don't line up with the scancode values of any specific platform, so there shouldn't be an use case where you need to convert numbers you get from elsewhere directly into them.

**Comment by kkimdev** [comment]
SDL returns what's already defined in this crate.  Here is the SDL's constant header file https://www.libsdl.org/tmp/SDL/include/SDL_scancode.h `SDL_SCANCODE_*` .

```C++
typedef enum
{
    SDL_SCANCODE_UNKNOWN = 0,

    /**
     *  \name Usage page 0x07
     *
     *  These values are from usage page 0x07 (USB keyboard page).
     */
    /* @{ */

    SDL_SCANCODE_A = 4,
    SDL_SCANCODE_B = 5,
    SDL_SCANCODE_C = 6,
    SDL_SCANCODE_D = 7,
    SDL_SCANCODE_E = 8,
    SDL_SCANCODE_F = 9,
    SDL_SCANCODE_G = 10,
    SDL_SCANCODE_H = 11,
    SDL_SCANCODE_I = 12,
    SDL_SCANCODE_J = 13,
    SDL_SCANCODE_K = 14,
    SDL_SCANCODE_L = 15,
...
```

**Comment by rsaarelm** [comment]
Hm, looks like SDL2 no longer shows you the underlying hardware scancode, just the remapping to the SDL2-specific enum. So yes, I see how this is a problem. Thing is, I'm not actually declaring any explicit API guarantees that my enum lines up with theirs, we're both using the USB HID document as source, but I'm considering of both enums as basically arbitrary and subject to change in the future.

If you're only using SDL2, do you even need `scancode-rs`? SDL2 is basically doing the exact same thing I am, so you should just use the scancode enum it gives you. If you're doing some sort of multi-backend thing, I'd just handle the SDL2 case by writing a big match statement with a line for each `scancode-rs` scancode, like

    match sdl2_scancode_enum {
        ...
        sdl2::keyboard::Scancode::A => Some(scancode_rs::Scancode::A),
        ...
        _ => None
    }

This will not mysteriously break down if either `scancode-rs` or SDL2 change their enum layout in the future.

**Comment by kkimdev** [comment]
OK, thanks!

**Comment by kkimdev** [comment]
SDL explicitly mentions that their scancode is based on the USB document https://wiki.libsdl.org/SDL_Scancode so I think I all I need is a scancode-stable rust crate if I want to convert the integer variable directly.

**Comment by rsaarelm** [comment]
I figured that I might as well support this. Released 0.1.1 with FromPrimitive support.

**Comment by kkimdev** [comment]
wow thanks!
